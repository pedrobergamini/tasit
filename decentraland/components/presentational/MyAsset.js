import React from "react";
import { StyleSheet, View } from "react-native";
import PropTypes from "prop-types";
import { responsiveWidth } from "react-native-responsive-dimensions";
import LinkToBlockchain from "./LinkToBlockchain";
import Estate from "./Estate";
import Parcel from "./Parcel";
import AssetTypes from "@constants/AssetTypes";
import AssetName from "./AssetName";

const { ESTATE, PARCEL } = AssetTypes;

export default function MyAsset({ asset }) {
  const { type, actionId } = asset;

  return (
    <View style={styles.assetContainer}>
      {(() => {
        switch (type) {
          case ESTATE:
            return <Estate estate={asset} />;
          case PARCEL:
            return <Parcel parcel={asset} />;
        }
      })()}
      <View style={styles.landInfoContainer}>
        <AssetName asset={asset} />
        <LinkToBlockchain actionId={actionId} />
      </View>
    </View>
  );
}

MyAsset.propTypes = {
  asset: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  landInfoContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  assetContainer: {
    width: responsiveWidth(95),
  },
});
