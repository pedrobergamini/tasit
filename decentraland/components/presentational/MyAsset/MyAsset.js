import React from "react";
import { StyleSheet, View } from "react-native";
import PropTypes from "prop-types";
import { responsiveWidth } from "react-native-responsive-dimensions";
import Estate from "@presentational/Estate";
import Parcel from "@presentational/Parcel";
import { ESTATE, PARCEL } from "@constants/AssetTypes";
import LinkToBlockchain from "@presentational/LinkToBlockchain";
import AssetName from "@presentational/AssetName";

export function MyAsset({ asset, userAction }) {
  const { type } = asset;

  return (
    <View style={styles.assetContainer}>
      {(() => {
        switch (type) {
          case ESTATE:
            return <Estate estate={asset} />;
          case PARCEL:
            return <Parcel parcel={asset} />;
        }
      })()}
      <MyAssetInfo asset={asset} userAction={userAction} />
    </View>
  );
}

MyAsset.propTypes = {
  asset: PropTypes.object.isRequired,
  userAction: PropTypes.object.isRequired,
};

export function MyAssetInfo({ asset, userAction }) {
  const { name } = asset;
  const [actionId] = Object.keys(userAction);

  return (
    <View style={styles.myAssetInfoContainer}>
      <View style={styles.nameContainer}>
        <AssetName name={name} />
      </View>
      <View style={styles.linkContainer}>
        <LinkToBlockchain actionId={actionId} />
      </View>
    </View>
  );
}

MyAssetInfo.propTypes = {
  asset: PropTypes.object.isRequired,
  userAction: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  assetContainer: {
    width: responsiveWidth(95),
  },
  linkContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-end",
  },
  myAssetInfoContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  nameContainer: {
    flex: 1,
  },
});

export default MyAsset;
