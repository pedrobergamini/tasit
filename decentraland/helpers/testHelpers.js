import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () =>
    "0x1234567890123456789012345678901234567890123456789012345678901234",
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
};
