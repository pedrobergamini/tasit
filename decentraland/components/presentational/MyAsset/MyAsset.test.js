import React from "react";
import { shallow } from "enzyme";
import { MyAsset, MyAssetInfo } from "./MyAsset";
// Not the @presentational version since it needs
// the internal component
import { parcel, parcelUserAction } from "@helpers/testHelpers";

describe("MyAsset", () => {
  it("renders the component", async () => {
    const asset = parcel;
    const userAction = parcelUserAction;

    expect(
      shallow(<MyAsset asset={asset} userAction={userAction} />)
    ).toMatchSnapshot();
  });

  describe("MyAssetInfo", () => {
    it("renders the component", async () => {
      const asset = parcel;
      const userAction = parcelUserAction;

      expect(
        shallow(<MyAssetInfo asset={asset} userAction={userAction} />)
      ).toMatchSnapshot();
    });
  });
});
