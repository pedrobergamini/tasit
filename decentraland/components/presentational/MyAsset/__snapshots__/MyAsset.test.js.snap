// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAsset MyAssetInfo renders the component 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <AssetName
      name="Sample Parcel"
    />
  </Component>
  <Component
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <LinkToBlockchain
      actionId="0x1234567890123456789012345678901234567890"
    />
  </Component>
</Component>
`;

exports[`MyAsset renders the component 1`] = `
<Component
  style={
    Object {
      "width": 712.5,
    }
  }
>
  <Parcel
    parcel={
      Object {
        "id": "0123456789",
        "img": "https://api.decentraland.org/v1/parcels/0/0/map.png",
        "name": "Sample Parcel",
        "type": "PARCEL",
      }
    }
  />
  <MyAssetInfo
    asset={
      Object {
        "id": "0123456789",
        "img": "https://api.decentraland.org/v1/parcels/0/0/map.png",
        "name": "Sample Parcel",
        "type": "PARCEL",
      }
    }
    userAction={
      Object {
        "0x1234567890123456789012345678901234567890": Object {
          "assetId": "0123456789",
          "status": "SUCCESSFUL",
        },
      }
    }
  />
</Component>
`;
